blocks/library/latest-posts/block.js

```diff
@@ -49,16 +49,17 @@ class LatestPostsBlock extends Component {
 
 	render() {
 		const latestPosts = this.props.latestPosts.data;
-		const { attributes, isSelected, setAttributes } = this.props;
+		const { attributes, categoriesList, isSelected, setAttributes } = this.props;
 		const { displayPostDate, align, layout, columns, order, orderBy, categories, postsToShow } = attributes;
 
 		const inspectorControls = isSelected && (
 			<InspectorControls key="inspector">
 				<h3>{ __( 'Latest Posts Settings' ) }</h3>
-				<QueryPanel
+				<QueryControls
 					{ ...{ order, orderBy } }
 					numberOfItems={ postsToShow }
-					category={ categories }
+					categoriesList={ get( categoriesList, 'data', {} ) }
```

Comment:
`categoriesList` will never be `undefined`, `get` isn't strictly necessary.

In a perfect world we wouldn't assume the source of the value, but given the `data` property access, we already kinda are.